x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/action_type_filter.tsx

```diff
@@ -5,42 +5,38 @@
  * 2.0.
  */
 
-import React, { useEffect, useState, useCallback } from 'react';
+import React, { useState, useCallback } from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { EuiPopover, EuiFilterButton, EuiFilterSelectItem } from '@elastic/eui';
 import { ActionType } from '../../../../types';
 
 interface ActionTypeFilterProps {
   actionTypes: ActionType[];
   onChange?: (selectedActionTypeIds: string[]) => void;
+  filters: string[];
 }
 
 export const ActionTypeFilter: React.FunctionComponent<ActionTypeFilterProps> = ({
   actionTypes,
-  onChange,
+  onChange: onFilterChange,
+  filters,
 }: ActionTypeFilterProps) => {
-  const [selectedValues, setSelectedValues] = useState<string[]>([]);
   const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
 
-  useEffect(() => {
-    if (onChange) {
-      onChange(selectedValues);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [selectedValues]);
-
   const onClick = useCallback(
     (item: ActionType) => {
       return () => {
-        const isPreviouslyChecked = selectedValues.includes(item.id);
+        if (!onFilterChange) return;
```

Comment:
`onFilterChange` is `setTypesFilter` here. 
It come from this line: `const [typesFilter, setTypesFilter] = useState<string[]>([]);`
Why do you check if it's falsy?

Reply:
It's not used as I see

Reply:
implemented here https://github.com/elastic/kibana/pull/144443/commits/fda6fba4dd1d8110ee658434f8038371877388fc